app/src/main/kotlin/com/hippo/ehviewer/ui/screen/GalleryDetailContent.kt

```diff
@@ -764,23 +767,14 @@ private fun getRatingText(rating: Float): Int = when ((rating * 2).roundToInt())
     else -> R.string.rating_none
 }
 
-private fun List<GalleryTagGroup>.getArtistTag(): String? {
-    for (tagGroup in this) {
-        val namespace = tagGroup.nameSpace
-        if (namespace == TagNamespace.Artist || namespace == TagNamespace.Cosplayer) {
-            return "$namespace:${tagGroup.tags[0].removePrefix("_")}"
-        }
-    }
-    return null
-}
+private fun List<GalleryTagGroup>.artistTag() = find { (ns, _) -> ns == TagNamespace.Artist || ns == TagNamespace.Cosplayer }?.let { (ns, tags) -> "$ns:${tags[0].text}" }
 
 context(Context)
 @Composable
-private fun GalleryDetail?.collectPreviewItems() = rememberInVM(this) {
-    val detail = this@collectPreviewItems ?: return@rememberInVM emptyFlow()
-    val pageSize = detail.previewList.size
-    val pages = detail.pages
-    val previewPagesMap = detail.previewList.associateBy { it.position } as MutableMap
+private fun GalleryDetail.collectPreviewItems() = rememberInVM {
```

Comment:
Preview list may change if gallery detail changes, and it should cause recompose.